### PR TITLE
fix(eslint-plugin): [no-redundant-type-constituents] differentiate a types-error any from a true any

### LINKED ISSUE #9214: Bug: [no-redundant-type-constituents] Differentiate a types-error any from a "true" any
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.5&fileType=.ts&code=C4TwDgpgBAogTnA9nAKuCBnKBeKA5RYAaQDtEB3EqAHygAYBuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1ukQBNYmvAIYtaRUrTIcmyfJXyxELVBkiJo0DtEjgwAXxB6gA&tsconfig=&tokens=false

### Repro Code

```TypeScript
type ErrorTypes = NotKnown | 0;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-redundant-type-constituents": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

The error message shouldn't phrase the `NotKnown` as "`any`", since it's really an "error" type. It's confusing to have it mentioned as `any`.

### Actual Result

```plaintext
'any' overrides all other types in this union type. 1:19 - 1:27
```

### Additional Info

Split out of #8304 <- #8231.

💖

### PR DESCRIPTION
**Pull request by y-hsgw**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9214
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

### CODE REVIEW

File: packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts [hunk 1]
```diff
@@ -102,6 +102,10 @@ function describeLiteralType(type: ts.Type): string {
     return type.value.toString();
   }
 
+  if (tsutils.isIntrinsicErrorType(type) && type.aliasSymbol) {
+    return type.aliasSymbol.escapedName.toString();
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts:106]: [Bug] This doesn't mention the type as being an "error typed value" the way the other rules do. The message you'd get for e.g. the `NotKnown | 0` test case is:

> ```plaintext
> 'NotKnown' overrides all other types in this union type.
> ```

The message should at least mention it being an error type. Maybe...

> ```plaintext
> 'NotKnown' is an `error` type, which acts as an `any` and overrides all other types in this union type.
> ```

...that's a little wordy, so if you can think of a smoother message, that'd be great. 😄

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for getting started on this! 🙌

File: packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts (hunk 1 above)
Comment (y-hsgw) [reply packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts:106]: Thank you for the review! I've updated the error message in df04797. Please check it.

File: packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts [hunk 2]
```diff
@@ -201,6 +205,7 @@ export default createRule({
       primitiveOverridden: `{{primitive}} is overridden by the {{literal}} in this intersection type.`,
       overridden: `'{{typeName}}' is overridden by other types in this {{container}} type.`,
       overrides: `'{{typeName}}' overrides all other types in this {{container}} type.`,
+      errorTypeOverrides: `'{{typeName}}' is an 'error' type that acts as 'any' and overrides all other types in this {{container}} type.`,
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts:208]: 💯 beautiful.

**Review by JoshuaKGoldberg — APPROVED** [review]
Steve Carell and Will Ferrell dressed in suits from the movie Anchorman, laughing and dancing happily

**Pull request merged**